src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonNode.cs

```diff
@@ -232,6 +234,124 @@ public JsonNode Root
             }
         }
 
+        /// <summary>
+        /// Creates a new instance of the <see cref="JsonNode"/>. All children nodes are recursively cloned.
+        /// </summary>
+        public JsonNode DeepClone()
+        {
+            if (this is JsonObject jObject)
+            {
+                return jObject.DeepCloneObject();
+            }
+            else if (this is JsonArray jArray)
+            {
+                return jArray.DeepCloneArray();
+            }
+            else
+            {
+                return AsValue().DeepCloneValue();
+            }
+        }
+
+        /// <summary>
+        /// Returns <see cref="JsonValueKind"/> of current instance.
+        /// </summary>
+        public JsonValueKind GetValueKind()
+        {
+            if (this is JsonObject)
+            {
+                return JsonValueKind.Object;
+            }
+            else if (this is JsonArray)
+            {
+                return JsonValueKind.Array;
+            }
+            else
+            {
+                return AsValue().GetInternalValueKind();
+            }
+        }
+
+        /// <summary>
+        /// Returns property name of the current node from the parent object.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        /// The current parent is not a <see cref="JsonObject"/>.
+        /// </exception>
+        public string GetPropertyName()
+        {
+            if (_parent is JsonObject jObject)
+            {
+                return jObject.GetPropertyName(this);
+            }
+
+            throw new InvalidOperationException(SR.Format(SR.NodeWrongType, nameof(JsonObject)));
```

Comment:
We generally avoid inlining `throw` statements in hot path methods, and hide these behind methods. I would recommend putting this behind a method in [ThrowHelper.Node.cs](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.Node.cs). 

I notice that existing methods already do this, so would it be possible to update these as well? Thanks.